src/app/nonprofit/nonprofit-page/nonprofit-page.component.ts

```diff
@@ -74,3 +90,95 @@ export class NonprofitPageComponent implements OnInit {
     console.log('nonprofit rating changed to: ' + rating);
   }
 }
+
+// ===================================================
+// Defining Component class for Dialog
+// ===================================================
+
+@Component({
+  selector: './donation-dialog',
+  templateUrl: './donation-dialog.html',
+  styleUrls: ['./donation-dialog.scss']
+})
+export class DonationDialog {
+  // Defining Step Forms
+  donationAmountForm: FormGroup;
+  donorInformationForm: FormGroup;
+  confirmationForm: FormGroup;
+
+  // Variables to be used in confirmation step
+  confCardHolderName: string;
+  confExpDate: string;
+  confCardNumber: string;
+  confCardCVC: string;
+  confDonationAmount: string;
+  
+
+  constructor(@Inject(MAT_DIALOG_DATA) public data: any, private formBuilder: FormBuilder, private dialogRef: MatDialog) {
+    // Initializing confirmation variables
+    this.confCardHolderName = "";
+    this.confExpDate = "";
+    this.confCardNumber = "";
+    this.confCardCVC = "";
+    this.confDonationAmount = "";
+
+    // Donation Amount Form
+    this.donationAmountForm = this.formBuilder.group({
+      donationAmount: ['', Validators.required]
+    });
+
+    // Donor Information Form
+    this.donorInformationForm = this.formBuilder.group({
+      cardHolderName: ['', Validators.required],
+      cardExpirationDate: ['', Validators.required],
+      cardNumber: ['', Validators.required],
+      cardCVC: ['', Validators.required]
+    });
+
+    // Confirmation Form
+    this.confirmationForm = this.formBuilder.group({
+      
+    });
+  }
+
+  submitDonation()
```

Comment:
[SUGGESTION] On submit, submit the payment, but delete the values immediately once submission succeeds,  from the variables (totally clear them). not sure how secure angular is in terms of storing variables or anything but best to be safe when handling credit card info and what not